pkg/services/webhook.go

```diff
@@ -91,10 +92,16 @@ type webhookService struct {
 }
 
 func (s webhookService) Send(notification Notification, dest Destination) error {
+	webhookUrl := s.opts.URL
+	_, urlParseError := url.ParseRequestURI(dest.Recipient)
```

Comment:
Looks like I was doing completely the same in https://github.com/argoproj/notifications-engine/pull/60 but for MS Teams. 
Could be useful to move check of a URL validity to some outside function and re-use everywhere, where an URL might appear :)

Reply:
yep, your real-world scenario is exactly the same!
i will have a look into your PR and try to find a place where this url-check can be put.

Reply:
@alexmt  looks like we can merge two PRs into something one combined with the same ideas. What do you think? :)

Actually, looks like only these two services can be upgraded with such a feature, but still moving out the check if the annotation is a URL somewhere outside of the service itself looks quite useful.

Reply:
Looks like pkg/util/misc/misc.go might be a good place to the function

Reply:
I've just updated my PR with the switch to enable/disable allowed URLs in the annotation. Might be useful to implement such an option here as well, @jimtoniq.